### PR TITLE
Bugfix/gal 1164

### PR DESCRIPTION
**Pull request by NeerjaGoyalNHS**
## Description
https://nhsd-jira.digital.nhs.uk/browse/GAL-1164

1. Invited and identified_to_be_invited fields were not there in Population DynamoDB 
2. Fields sent by Caas are named differently from fields in testdata on basis of which coding has been done. Both has been aligned as per Caas Feed. 

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/getLsoaParticipantsLambda/getLsoaParticipantsLambda.js [hunk 1]
```diff
@@ -116,9 +116,9 @@ export async function getEligiblePopulation(lsoaList, client) {
           !person?.identified_to_be_invited?.BOOL &&
           person?.Invited?.S == "false" &&
           person?.date_of_death?.S == "NULL" &&
-          person?.removal_date?.S == "NULL" &&
+          person?.reason_for_removal_effective_from_date?.S == "NULL" &&
           person?.superseded_by_nhs_number?.N == 0 &&
-          person?.removal_reason?.S == "NULL"
+          person?.reason_for_removal?.N == "NULL"
```
Comment (chenglawson1) [inline terraform/src/getLsoaParticipantsLambda/getLsoaParticipantsLambda.js:121]: reason_for_removal is of type string not a number.

File: terraform/main.tf [hunk 2]
```diff
@@ -1684,7 +1684,7 @@ module "population_table" {
   secondary_write_capacity = null
   secondary_read_capacity  = null
   environment              = var.environment
-  # non_key_attributes     = ["Invited", "date_of_death", "removal_date", "identified_to_be_invited", "LsoaCode", "postcode", "PersonId", "primary_care_provider"]
+  # non_key_attributes     = ["Invited", "date_of_death", "reason_for_removal_effective_from_date", "identified_to_be_invited", "LsoaCode", "postcode", "PersonId", "primary_care_provider"]
```
Comment (andrew-cleveland) [inline terraform/main.tf:1687]: You seem to have updated a commented out section here, should this be uncommented to be used or should it be removed if unnecessary?

Comment (NeerjaGoyalNHS) [reply terraform/main.tf:1687]: safe sided I updated the commented out section as well.... as I am of understanding that commented section in .tf files might be used in future.....

Comment (NeerjaGoyalNHS) [reply terraform/main.tf]: removed.. thanks

File: terraform/src/getLsoaParticipantsLambda/getLsoaParticipantsLambda.js (hunk 1 above)
Comment (NeerjaGoyalNHS) [reply terraform/src/getLsoaParticipantsLambda/getLsoaParticipantsLambda.js]: Changed to string... Thanks

**Review by Abdul-NHS — DISMISSED** [review]

**Review by andrew-cleveland — APPROVED** [review]

**Review by Amar-NHS — APPROVED** [review]

**Pull request merged**